server.js

```
@@ -0,0 +1,218 @@
+const http = require("http");
+const fs = require("fs");
+const querystring = require("querystring");
+const PORT = 8080;
+let date = new Date().toUTCString();
+
+const server = http.createServer((req, res) => {
+  // Body takes in chunks requested by client
+  let body = "";
+  req.on("data", chunk => {
+    body += chunk;
+  });
+  req.on("end", () => {
+    /*** GET Method ***/
+    if (req.method === "GET") {
+      // Grabs files from public to show to client
+      let directory;
+      let contentType;
+      fs.readdir('./public/', (err, files) => {
+        if (files.includes(req.url.slice(1)) === false) { // Directory doesn't exist
+          if (req.url === "/") { // For default page
+            directory = "/index.html";
+            contentType = "html"
+          } else if (req.url.slice(-3) === "css") { // Needed to load css properly
+            directory = `${req.url}`
+            contentType = "css"
+          } else {
+            directory = '/404.html';
+            contentType = 'html';
+          }
+        } else { // Directory exists
+          if (req.url.slice(-3) === "css") { // Needed to load css properly
+            directory = `${req.url}`
+            contentType = "css"
+          } else {
+            directory = req.url
+            contentType = req.url.slice(req.url.indexOf('.') + 1, req.url.length);
+          }
+        }
+        fs.readFile(`./public${directory}`, (err, data) => { // Sends requested data
+          if (err) {
+            return console.log(err);
```

This should return an error to the client using either a 400/500 error